pkg/providers/tinkerbell/upgrade.go

```diff
@@ -506,3 +512,52 @@ func toRufioMachines(items []rufiounreleased.BaseboardManagement) []rufiov1.Mach
 	}
 	return machines
 }
+
+// IsRollingNotScale compares current CAPI objects and desired CAPI objects
+// returns true on rolling upgrade, false on scale upgrade.
+// should error if rolling and scale at the same time
+// todo: IsRollingNotScale
+func IsRollingNotScale(ctx context.Context, log logr.Logger, client runtimeclient.Client, clusterSpec *cluster.Spec) (bool, error) {
+	cp, err := ControlPlaneSpec(ctx, log, clientutil.NewKubeClient(client), clusterSpec)
+	if err != nil {
+		return false, err
+	}
+
+	w, err := WorkersSpec(ctx, log, clientutil.NewKubeClient(client), clusterSpec)
+	if err != nil {
+		return false, err
+	}
+
+	currentKCP := &controlplanev1.KubeadmControlPlane{}
+	err = client.Get(ctx, runtimeclient.ObjectKey{
+		Name:      cp.KubeadmControlPlane.Name,
+		Namespace: cp.KubeadmControlPlane.Namespace,
+	}, currentKCP)
+	if apierrors.IsNotFound(err) {
```

Comment:
Can we return false and err here? This could be a better place to place cluster existence check, to remove duplicate logic in reconciler/cp/worker. I'll update my pr and check error for `IsNotFound`.